core/mondoo-aws-security.mql.yaml

```diff
@@ -2265,7 +2265,7 @@ queries:
     filters: |
       asset.platform == "aws-sagemaker-notebookinstance"
     mql: |
-      aws.sagemaker.notebookInstance
+      aws.sagemaker.notebookInstances
```

Comment:
but don't we need the singular here, not the list of instances? i think it needs to be `notebookinstance`

Reply:
Generally speaking, @tas50 original suggestion is the way to go.

```
aws.sagemaker.notebookinstance.details.kmsKey != empty
```

But if the singular resource doesn't [work](https://github.com/mondoohq/cnspec-policies/pull/339#discussion_r1462184734) [properly](https://github.com/mondoohq/cnquery/issues/3032), it is a good workaround to limit the number of assets on the general resource with a where-clause until it does work.

```
aws.sagemaker.notebookInstances
  .where(asset.name == name)
  .all(details.kmsKey != empty)
```

Reply:
do we know the single one is not working properly? the example above doesn't show all lowercase notebookinstance, it's camel case